### DISCUSSION TITLE
How to set vscode server location and terminal profile?

### DISCUSSION THREAD

**Discussion by uduse** [devcontainers/community#7] [Q&A] [answered]
In my Dockerfile I specify `USER root`. In my `.devcontainer.json` I set `containerUser": "root"`. Once I build the container and open it with vscode, I create a new terminal and this is the prompt:
```bash
bash: /home/ubuntu/.bashrc: No such file or directory
root@6cbd5f5a991c:/workspace# 
```
I tried to set the user to root whenever possible. What am I missing here?

**Comment by Chuxel** [comment, marked as answer]
@uduse What does your Dockerfile and .devcontainer.json look like?  Why do you think this is related to the vscode server location or terminal profile?

**Reply by uduse** [reply]
My Dockerfile and `.devcontainer.json` can be found [here](https://github.com/niftylab/laygo2_workspace_sky130/pull/8/files#diff-f6b691a938d86a465fe65f961c97f11ebb3013e6c0639385f8fda1fe6585e4ad). I think they are related because I eyeballed the logs of Dev Container starting up and it seems these `/home/ubuntu/` locations are used for server location. For the terminal profile, it's also just a guess.

**Reply by Chuxel** [reply]
The base image you are using appears hard code `HOME` to that location along with the shell.  

https://github.com/fcwu/docker-ubuntu-vnc-desktop/blob/develop/Dockerfile.amd64

```
ENV HOME=/home/ubuntu \
        SHELL=/bin/bash
```

The `USER` and `PASSWORD` environment variables described for the image are not related to Dev Containers but rather part of the image itself.

The `$HOME` directory is where things would be installed. In this case, the image author is dictating that location.  

You likely also want to set `"overrideCommand": false` given this has a startup script.  It's possible that is populating the bashrc file. It defaults to being overridden since many images exit immediately by default.

**Reply by uduse** [reply]
@Chuxel Thanks a lot for the answer! This is very helpful 👍
